internal/mutator/pod/agent/injector.go

```diff
@@ -439,34 +445,63 @@ func (i *injector) ensureSidecarDependencies(ctx context.Context, pod *corev1.Po
 	return nil
 }
 
-func (i *injector) computeResourcesToApply(podLabels map[string]string) *corev1.ResourceRequirements {
+func (i *injector) applyAgentConfig(podLabels map[string]string, container *corev1.Container) string {
 	for _, r := range i.config.AgentConfig.ConfigSelectors {
 		if r.selector.Matches(labels.Set(podLabels)) {
-			return r.ResourceRequirements
-		}
-	}
+			if r.ResourceRequirements != nil {
+				container.Resources = *r.ResourceRequirements
+			}
 
-	return nil
-}
+			for k, v := range r.ExtraEnvVars {
+				container.Env = append(container.Env, corev1.EnvVar{Name: k, Value: v})
+			}
 
-func (i *injector) computeEnvVarsToApply(podLabels map[string]string) map[string]string {
-	for _, r := range i.config.AgentConfig.ConfigSelectors {
-		if r.selector.Matches(labels.Set(podLabels)) {
-			return r.ExtraEnvVars
+			return r.hash
 		}
 	}
 
-	return nil
+	return i.configHash
+}
+
+type configHash struct {
+	CustomAttributes     CustomAttributes
+	ResourcePrefix       string
+	ClusterName          string
+	Image                Image
+	PodSecurityContext   PodSecurityContext
+	ResourceRequirements *corev1.ResourceRequirements
+	ExtraEnvVars         map[string]string
+	Container            corev1.Container
 }
 
-func (config *InjectorConfig) buildConfigSelectors() error {
+func (config *InjectorConfig) buildConfigSelectors(container corev1.Container, logger *logrus.Logger) error {
 	for i, r := range config.AgentConfig.ConfigSelectors {
 		selector, err := metav1.LabelSelectorAsSelector(&r.LabelSelector)
 		if err != nil {
 			return fmt.Errorf("creating selector from label selector: %w", err)
 		}
 
 		config.AgentConfig.ConfigSelectors[i].selector = selector
+
+		configHash := &configHash{
+			CustomAttributes:     config.AgentConfig.CustomAttributes,
+			ResourcePrefix:       config.ResourcePrefix,
+			ClusterName:          config.ClusterName,
+			Image:                config.AgentConfig.Image,
+			PodSecurityContext:   config.AgentConfig.PodSecurityContext,
+			ResourceRequirements: r.ResourceRequirements,
+			ExtraEnvVars:         r.ExtraEnvVars,
+			Container:            container,
+		}
+
+		hash, err := configHash.calculate()
+		if err != nil {
+			return fmt.Errorf("calculating config hash: %w", err)
+		}
+
+		logger.Infof("'%s' label value for Pod with config selector %d: '%s'", InjectedLabel, i, hash)
```

Comment:
Therefore is there a 1to1 correspondence between hash and config selector?

Instead of having 1 hash we are going to have n valid ones I guess

Reply:
Yes. each config selector has it's own hash.